osbenchmark/synthetic_data_generator/helpers.py

```diff
@@ -0,0 +1,120 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+import os
+import logging
+import json
+
+import yaml
+
+from osbenchmark.utils import console
+from osbenchmark.exceptions import SystemSetupError
+from osbenchmark.synthetic_data_generator.types import DEFAULT_GENERATION_SETTINGS, SyntheticDataGeneratorConfig
+
+def load_config(config_path):
+    try:
+        allowed_extensions = ['.yml', '.yaml']
+
+        extension = os.path.splitext(config_path)[1]
+        if extension not in allowed_extensions:
+            raise SystemSetupError(f"User provided config with extension [{extension}]. Config must have a {allowed_extensions} extension.")
+
+        if config_path.endswith(allowed_extensions[0]) or config_path.endswith(allowed_extensions[1]):
+            with open(config_path, 'r') as file:
+                return yaml.safe_load(file)
+    except TypeError:
+        raise SystemSetupError("Error when loading config. Please ensure that the proper config was provided")
+
+def write_chunk(data, file_path):
+    with open(file_path, 'a') as f:
+        for item in data:
+            f.write(json.dumps(item) + '\n')
+    return len(data)
+
+def get_generation_settings(input_config: dict) -> dict:
+    '''
+    Grabs the user's config's generation settings and compares it with the default generation settings.
+    If there are missing fields in the user's config, it populates it with the default values
+    '''
+    generation_settings = DEFAULT_GENERATION_SETTINGS
```

Comment:
is this a strict schema or users can add other keys in the settings?
I think it is better to enforce else have better error handling for key error.

Reply:
This is a strict schema at the moment. Even if users define their own keys, this helper function only cycles through the valid keys, which have to exist in DEFAULT_GENERATION_SETTINGS.